### PR TITLE
Medtronic 5/7-series fixes

### PR DESCRIPTION
**Pull request by gniezen**
See attached Trello cards for details.

### CODE REVIEW

**Comment by gniezen** [comment]
[Error Details: Expected a normal bolus, but found type 4 (or 5)](https://trello.com/c/3zeww3MJ/427-error-details-expected-a-normal-bolus-but-found-type-4-or-5)

**Comment by gniezen** [comment]
[[uploader] Medronic Direct: No rate for temp basal (Uploading data to platform failed)](https://trello.com/c/RVSAh1aT/426-uploader-medronic-direct-no-rate-for-temp-basal-uploading-data-to-platform-failed)

**Comment by gniezen** [comment]
This now also contains the changes from #633 that were already reviewed, as they will be tested together.

File: lib/drivers/medtronic/processData.js [hunk 1]
```diff
@@ -335,73 +346,72 @@ var buildBolusRecords = function (records) {
...
       bolus = cfg.builder.makeNormalBolus()
         .with_normal(amount);
 
       if(programmed !== amount) {
         bolus = bolus.with_expectedNormal(programmed);
       }
+    } else {
+
+      if (subType !== BOLUS_TYPES.SQUARE) {
+        throw new Error('Expected a square bolus, but found type' + subType);
```
Comment (pazaan) [inline lib/drivers/medtronic/processData.js]: Add a space before the last quote.

**Comment by gniezen** [comment]
#633 is already merged into this PR. I'm going to also merge #628 and release it as a single update.

**Pull request merged**